Duvall Pinkney homework number one 
13128408
2/8/2021 spring 2021


1900	170
1910	172.1
1920	173.1
1930	173.4
1940	176.1
1950	177.1
1960	177.3
1970	178.3
1980	179


The above table contains the average height for men in the United States between 1900 and 1980. (Source: https://ourworldindata.org/human-height)

Write a Python program to convert the percent of the increase of men's height over each decade. For example, during 1900 - 1910 the average height has increased:

(172.1 - 170) / 170 = 1.235%.

Moreover, determine on during which 10 years has men's height increased the fastest.

Create a notebook file with code and execution results showing the percentages and the decade of the fastest growth. Save the notebook as a PDF file. You need to submit the PDF file to Blackboard before the deadline to receive grades.

In [1]:
import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sns
# The following line is needed for some versions of Python and Jupyter Notebooks to display the plots in the notebook.
%matplotlib inline

In [29]:
def men_height(earlyHeight, laterHeight):
    percentOfIncrease = (laterHeight - earlyHeight) / earlyHeight * 100
    return percentOfIncrease
    

In [3]:
height_data = pd.read_csv("mensHeightdata.csv")
height_data

1900    170
0  1910  172.1
1  1920  173.1
2  1930  173.4
3  1940  176.1
4  1950  177.1
5  1960  177.3
6  1970  178.3
7  1980  179.0

In [30]:

first_decade  = men_height(170,172.1)
print("1900 - 1910 percentage of increase is:", first_decade)



1900 - 1910 percentage of increase is: 1.2352941176470555


In [32]:
second_decade  = men_height(172.1,173.1)
print("1910 - 1920 percentage of increase is:", second_decade)

1910 - 1920 percentage of increase is: 0.5810575246949448


In [33]:
third_decade  = men_height(173.1,173.4)
print("1920 - 1930 percentage of increase is:", third_decade)

1920 - 1930 percentage of increase is: 0.17331022530329948


In [34]:
fourth_decade  = men_height(173.4,176.1)
print("1930 - 1940 percentage of increase is:", fourth_decade)

1930 - 1940 percentage of increase is: 1.5570934256055295


In [35]:
fifth_decade  = men_height(176.1,177.1)
print("1940 - 1950 percentage of increase is:", fifth_decade)

1940 - 1950 percentage of increase is: 0.5678591709256104


In [36]:
sixth_decade  = men_height(177.1,177.3)
print("1950 - 1960 percentage of increase is:", sixth_decade)

1950 - 1960 percentage of increase is: 0.11293054771316605


In [37]:
seventh_decade  = men_height(177.3,178.3)
print("1960 - 1970 percentage of increase is:", seventh_decade)

1960 - 1970 percentage of increase is: 0.5640157924421884


In [38]:
eighth_decade  = men_height(178.3,179.0)
print("1970 - 1980 percentage of increase is:", eighth_decade)

1970 - 1980 percentage of increase is: 0.39259674705551795


1930 - 1940 percentage of increase is: 1.5570934256055295
this increased the most.